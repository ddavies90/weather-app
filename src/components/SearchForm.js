import React from "react";
import PropTypes from "prop-types";
import "../styles/SearchForm.css";

const SearchForm = ({ searchValue, setSearchValue, citySearchFunc }) => {
  const handleInputChange = (event) => {
    setSearchValue(event.target.value);
  };

  return (
    <div className="search-form">
      <label htmlFor="search">
        <input
          type="text"
          name="search"
          placeholder="Enter a city name"
          onChange={handleInputChange}
          value={searchValue}
        />
      </label>
      <button type="submit" onClick={citySearchFunc}>
        Search
      </button>
    </div>
  );
};

SearchForm.propTypes = {
  searchValue: PropTypes.string.isRequired,
  setSearchValue: PropTypes.func.isRequired,
  citySearchFunc: PropTypes.func.isRequired
};

export default SearchForm;
