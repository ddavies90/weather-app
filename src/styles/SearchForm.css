.search-form {
  align-self: center;
}

.search-form button {
  margin-left: 0.7rem;
}

.search-form button:hover {
  background-color: red;
  color: yellow;
  transform: scale(1.2);
  box-shadow: 0 0 0 2px blue;
}

.search-form input {
  font-family: "VT323", monospace;
  font-size: 1.4rem;
  border: none;
  width: 12rem;
  padding: 0.3rem;
  background-color: blue;
}

.search-form input::placeholder {
  color: #dce6f5;
  opacity: 1;
}

.search-form input:focus,
.search-form button:focus {
  outline: 2px solid red;
}
