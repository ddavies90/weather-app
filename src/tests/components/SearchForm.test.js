import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("<SearchForm />", () => {
  const validProps = {
    searchValue: "Manchester",
    setSearchValue: jest.fn(),
    citySearchFunc: jest.fn()
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchValue={validProps.searchValue}
        setSearchValue={validProps.setSearchValue}
        citySearchFunc={validProps.citySearchFunc}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders search box and search button to be rendered with correct placeholder or text", () => {
    render(
      <SearchForm
        searchValue={validProps.searchValue}
        setSearchValue={validProps.setSearchValue}
        citySearchFunc={validProps.citySearchFunc}
      />
    );

    expect(
      screen.getByPlaceholderText("Enter a city name")
    ).toBeInTheDocument();

    expect(screen.getByText("Search")).toBeInTheDocument();
  });

  it("Calls citySearchFunc when the search button is clicked or enter is pressed on the text input", () => {
    render(
      <SearchForm
        searchValue={validProps.searchValue}
        setSearchValue={validProps.setSearchValue}
        citySearchFunc={validProps.citySearchFunc}
      />
    );

    fireEvent.click(screen.getByText("Search"));
    fireEvent.keyDown(screen.getByRole("textbox"), { key: "Enter" });

    expect(validProps.citySearchFunc).toBeCalledTimes(2);
  });

  it("value updates when text input is changed", () => {
    render(
      <SearchForm
        searchValue={validProps.searchValue}
        setSearchValue={validProps.setSearchValue}
        citySearchFunc={validProps.citySearchFunc}
      />
    );

    fireEvent.change(screen.getByRole("textbox"), {
      target: { value: "Manchester" }
    });

    expect(screen.getByRole("textbox").value).toEqual("Manchester");
  });
});
