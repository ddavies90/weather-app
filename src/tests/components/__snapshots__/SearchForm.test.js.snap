// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <label
      for="search"
    >
      <input
        name="search"
        placeholder="Enter a city name"
        type="text"
        value="Manchester"
      />
    </label>
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
